### PR TITLE
add folding markers for @doc, @moduledoc, #region

### PR DESCRIPTION
**Pull request by lastobelus**

### CODE REVIEW

**Comment by lastobelus** [comment]
See #156

**Comment by lastobelus** [comment]
It would probably be better if `@doc` and `@moduledoc` got the textmate scope `comment.block` in addition to `comment.documentation`, then I believe the built in folding of block comments, with the special `Fold all block comments` command would work.

File: elixir-language-configuration.json [hunk 1]
```diff
@@ -59,5 +59,11 @@
   "indentationRules": {
     "increaseIndentPattern": "(after|else|catch|rescue|fn|^.*(do|<\\-|\\->|\\{|\\[|\\=))\\s*$",
     "decreaseIndentPattern": "^\\s*((\\}|\\])\\s*$|(after|else|catch|rescue|end)\\b)"
+  },
+  "folding": {
+    "markers": {
+      "start": "^\\s*(@(moduledoc|doc)\\b\\s+\"\"\")|(#\\s*region\\b)",
```
Comment (axelson) [inline elixir-language-configuration.json:65]:
```suggestion
      "start": "^\\s*(@(moduledoc|typedoc|doc)\\b\\s+\"\"\")|(#\\s*region\\b)",
```

**Review by axelson — APPROVED** [review]
Thanks :heart: , just added `typedoc` support as well

**Comment by axelson** [comment]
And feel free to open another PR that changes/adds `comment.block` for type/module/docs

**Comment by jbosse** [comment]
The doc folding seems to work, however I cannot seem to get a region to fold.
```elixir
# region
def my_method do
  #...
end
# end region
```
@lastobelus This is how it is supposed to be marked, correct?

**Pull request merged**